# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 1bdbc884e289
# Full Hash: 1bdbc884e289f2656391c85a1fb35126c4e21ad8
# Author: Perry Jiang <perry@mozilla.com>
# Date: 2020-07-15 16:06:28
# Description:
#   Bug 1632036 - check for null global when freezing/thawing WorkerPrivate r=dom-workers-and-storage-reviewers,sg
#   
#   - Apply pointer guidelines to ClientSource getter.
#   - When freezing/thawing WorkerPrivate, check if the client/global is null. It seems possible that the client is null because the WorkerPrivate can remain registered to the RuntimeService even if it failed to create its global, and freezing/thawing is done through the RuntimeService. The other two callsites to get the client won't execute unless CompileScriptRunnable successfully creates the client.
#   
# ==============================================================================

diff -r 4e76e9b8d9e5 -r 1bdbc884e289 dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Jul 15 04:40:08 2020 +0000
+++ b/dom/workers/WorkerScope.h	Thu Jul 09 07:25:16 2020 +0000
@@ -130,9 +130,7 @@
 
   void NoteTerminating() { StartDying(); }
 
-  const UniquePtr<ClientSource>& GetClientSource() const {
-    return mClientSource;
-  }
+  ClientSource& MutableClientSourceRef() const { return *mClientSource; }
 
   // WorkerPrivate wants to be able to forbid script when its state machine
   // demands it.
